allyabase-web/script.js

```diff
@@ -0,0 +1,162 @@
+                if (serviceName && cachedDescriptions[serviceName]) {
+                    const tooltip = document.createElement('div');
+                    tooltip.className = 'tooltip';
+                    tooltip.textContent = cachedDescriptions[serviceName];
+                    shape.appendChild(tooltip);
+                }
+            });
+            return;
+        }
+
+        // If no cache, fetch from GitHub API
+        const descriptions = {};
+        const fetchPromises = [];
+
+        shapes.forEach(shape => {
+            const serviceName = shape.getAttribute('data-service');
+            if (serviceName && serviceUrls[serviceName]) {
+                const repoUrl = serviceUrls[serviceName];
+                const repoName = repoUrl.split('/').pop();
+                
+                const fetchPromise = fetch(`https://api.github.com/repos/planet-nine-app/${repoName}`)
+                    .then(response => response.json())
+                    .then(data => {
+                        descriptions[serviceName] = data.description || 'No description available';
+                        
+                        const tooltip = document.createElement('div');
+                        tooltip.className = 'tooltip';
+                        tooltip.textContent = descriptions[serviceName];
+                        shape.appendChild(tooltip);
+                    })
+                    .catch(error => {
+                        console.error(`Error fetching description for ${serviceName}:`, error);
+                        descriptions[serviceName] = 'Description unavailable';
+                    });
+                
+                fetchPromises.push(fetchPromise);
+            }
+        });
+
+        // Wait for all fetches to complete then cache the results
+        try {
+            await Promise.all(fetchPromises);
+            setCache(descriptions);
+        } catch (error) {
+            console.error('Error caching descriptions:', error);
+        }
+    };
+
+    // Initialize tooltips
+    fetchRepoDescriptions();
+
+    // Handle gem and shape clicks
+    shapesContainer.addEventListener('click', (e) => {
+        if (!isExpanded) {
+            // First click: expand the gem into shapes
+            shapesContainer.classList.add('expanded');
+            isExpanded = true;
+            // Disable parallax when expanded
+            document.removeEventListener('mousemove', handleParallax);
+        }
+    });
+
+    shapes.forEach(shape => {
+        shape.addEventListener('click', (e) => {
+            if (isExpanded) {
+                e.stopPropagation(); // Prevent container click
+                // Navigate to service repo
+                const serviceName = shape.getAttribute('data-service');
+                if (serviceUrls[serviceName]) {
+                    window.location.href = serviceUrls[serviceName];
+                }
+            }
+        });
+    });
```

Comment:
I don't see anything wrong with this, but would it be easier to just wrap the shapes in a tags with the serviceURLs?